### PR TITLE
Use secure traceback/logging

### PR DESCRIPTION
**Pull request by YuanTingHsieh**
- Use a variable "NVFLARE_SECURE_LOGGING" to control the details of stack trace

### CODE REVIEW

**Comment by YuanTingHsieh** [comment]
will make some changes based on feedback from @yanchengnv .
Basically to control whether print the exception message or not.

File: nvflare/apis/fl_component.py [hunk 1]
```diff
@@ -213,10 +213,10 @@ def log_exception(self, fl_ctx: FLContext, msg: str, fire_event=False):
         """
         log_msg = generate_log_message(fl_ctx, msg)
```
[the remark this answers is not in the document]
Comment (YuanTingHsieh) [reply nvflare/apis/fl_component.py:214]: I can't find where `generate_log_message` prints out the shareable dict, can you point that out, thanks

File: nvflare/app_common/homomorphic_encryption/he_model_encryptor.py [hunk 2]
```diff
@@ -185,7 +185,7 @@ def process(self, shareable: Shareable, fl_ctx: FLContext) -> Shareable:
         try:
             new_dxo = self._process(dxo, fl_ctx)
             new_dxo.update_shareable(shareable)
-        except BaseException as e:
+        except BaseException:
```
[the remark this answers is not in the document]
Comment (YuanTingHsieh) [reply nvflare/app_common/homomorphic_encryption/he_model_encryptor.py:188]: I was trying to minimize the changes, if we want to print the exception sure I can change all the places.

Note that these changes are not directly applicable to 2.1 or 2.2, they need to be done separately.

File: nvflare/apis/fl_component.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (YuanTingHsieh) [reply nvflare/apis/fl_component.py:214]: the naming is poor, if you look at all the codes only run_number, workflow name, peer name, peer run number, return code, task name and task id is added into this "ctx_items".

these infos are not sensitive IMO

```
https://github.com/YuanTingHsieh/NVFlare/blob/1fd3294140db10e259ece51320b95336a8914ed0/nvflare/apis/utils/fl_context_utils.py#L83
```

[the remark this answers is not in the document]
Comment (yanchengnv) [reply nvflare/apis/fl_component.py:214]: Agree with Yuan-Ting.

File: nvflare/app_common/homomorphic_encryption/he_model_encryptor.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (yanchengnv) [reply nvflare/app_common/homomorphic_encryption/he_model_encryptor.py:188]: Let's minimize the scope of this - do not change anything else for 2.0. We can improve it in 2.2 if necessary.

**Pull request merged**